Installation issue: openssh %nvhpc

### Steps to reproduce the issue

$ spack maintainers openssh
@bernhardkaindl

$ spack install openssh %nvhpc@22.1 ^openssl %gcc
```
[+] /home2/cessenat/Softs/spack/opt/spack/linux-ubuntu20.04-zen2/nvhpc-22.1/pkgconf-1.8.0-a3kn22sdo3dikakyqkux3xsr4gcmnu5n
[+] /usr (external perl-5.30.0-gmkstxoj5w7t5eemib4yd365zw3nqayk)
[+] /home2/cessenat/Softs/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/zlib-1.2.11-whj3n7mok67wqtm5dofead56i6krevxc
[+] /home2/cessenat/Softs/spack/opt/spack/linux-ubuntu20.04-zen2/nvhpc-22.1/ncurses-6.2-7yglxp6sw2ajhjpdgaxfs63hu3i2dok7
[+] /home2/cessenat/Softs/spack/opt/spack/linux-ubuntu20.04-zen2/gcc-9.3.0/openssl-1.1.1m-23imk2rsftvme5mwrkr6wdvhoecqq2rk
[+] /home2/cessenat/Softs/spack/opt/spack/linux-ubuntu20.04-zen2/nvhpc-22.1/libedit-3.1-20210216-c5mkfnd4gab6cuceloxorgpypr565b5u
==> Installing openssh-8.8p1-m6zc5n3ra3rfeg4senjydunhcsq6ub56
==> No binary for openssh-8.8p1-m6zc5n3ra3rfeg4senjydunhcsq6ub56 found: installing from source
==> Using cached archive: /home2/cessenat/Softs/spack/var/spack/cache/_source-cache/archive/45/4590890ea9bb9ace4f71ae331785a3a5823232435161960ed5fc86588f331fe9.tar.gz
==> No patches needed for openssh
==> openssh: Executing phase: 'autoreconf'
==> openssh: Executing phase: 'configure'
==> openssh: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j6' 'V=1'

2 errors found in build log:
     6092    a - sshbuf-io.o
     6093    ranlib libssh.a
     6094    /home2/cessenat/Softs/spack/lib/spack/env/nvhpc/nvc -o ssh ssh.o readconf.o clientloop.o sshtty.o sshconnect.o sshconnect2.o mux.o ssh-sk-c
             lient.o -L. -Lopenbsd-compat/  -Wl,-z,relro -Wl,-z,now -Wl,-z,noexecstack -fstack-protector-strong -pie  -lssh -lopenbsd-compat -lcrypto -l
             util -lz  -lcrypt -lresolv
     6095    nvc-Warning-Unknown switch: -fstack-protector-strong
     6096    /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/crtbegin.o: relocation R_X86_64_32 against hidden symbol `__TMC_END__' can not be used when ma
             king a PIE object
     6097    /home2/cessenat/Softs/spack/lib/spack/env/nvhpc/nvc -o sshd sshd.o auth-rhosts.o auth-passwd.o audit.o audit-bsm.o audit-linux.o platform.o
              sshpty.o sshlogin.o servconf.o serverloop.o auth.o auth2.o auth-options.o session.o auth2-chall.o groupaccess.o auth-bsdauth.o auth2-hostb
             ased.o auth2-kbdint.o auth2-none.o auth2-passwd.o auth2-pubkey.o monitor.o monitor_wrap.o auth-krb5.o auth2-gss.o gss-serv.o gss-serv-krb5.
             o loginrec.o auth-pam.o auth-shadow.o auth-sia.o md5crypt.o srclimit.o sftp-server.o sftp-common.o sandbox-null.o sandbox-rlimit.o sandbox-
             systrace.o sandbox-darwin.o sandbox-seccomp-filter.o sandbox-capsicum.o sandbox-pledge.o sandbox-solaris.o uidswap.o ssh-sk-client.o -L. -L
             openbsd-compat/  -Wl,-z,relro -Wl,-z,now -Wl,-z,noexecstack -fstack-protector-strong -pie  -lssh -lopenbsd-compat  -lcrypto -lutil -lz  -lc
             rypt -lresolv
  >> 6098    make: *** [Makefile:210: ssh] Error 2
     6099    make: *** Waiting for unfinished jobs....
     6100    nvc-Warning-Unknown switch: -fstack-protector-strong
     6101    /usr/bin/ld: /usr/lib/gcc/x86_64-linux-gnu/9/crtbegin.o: relocation R_X86_64_32 against hidden symbol `__TMC_END__' can not be used when ma
             king a PIE object
  >> 6102    make: *** [Makefile:213: sshd] Error 2

See build log for details:
  /tmp/cessenat/spack-stage/spack-stage-openssh-8.8p1-m6zc5n3ra3rfeg4senjydunhcsq6ub56/spack-build-out.txt

```

### Information on your system

$ spack debug report
* **Spack:** 0.17.0-856-e0f044561e
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-zen2
* **Concretizer:** clingo


### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/7883866/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/7883867/spack-build-out.txt)


### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/openssh/package.py
+++ b/var/spack/repos/builtin/packages/openssh/package.py
@@ -66,6 +66,10 @@ class Openssh(AutotoolsPackage):
         # least newer versions want to create the directory during the
         # install step and fail if they cannot do so.
         args = ['--with-privsep-path={0}'.format(self.prefix.var.empty)]
+
+        # Somehow creating pie executables fails with nvhpc, not with gcc.
+        if '%nvhpc' in self.spec:
+            args.append('--without-pie')
         return args
 
     def install(self, spec, prefix):

